src/sentry/migrations/0024_auto_20191230_2052.py

```diff
@@ -69,24 +70,33 @@ def _attach_related(_events):
 
     processed = 0
     for e in RangeQuerySetWrapper(events, step=100, callbacks=(_attach_related,)):
-        event = NewEvent(project_id=e.project_id, event_id=e.event_id, group_id=e.group_id, data=e.data.data)
+        event = NewEvent(
+            project_id=e.project_id, event_id=e.event_id, group_id=e.group_id, data=e.data.data
+        )
         primary_hash = event.get_primary_hash()
         if event.project is None or event.group is None:
             print("Skipped {} as group or project information is invalid.\n".format(event))
             continue
 
-        eventstream.insert(
-            group=event.group,
-            event=event,
-            is_new=False,
-            is_regression=False,
-            is_new_group_environment=False,
-            primary_hash=primary_hash,
-            skip_consume=True,
-        )
-        processed += 1
-
-    print("Event migration done. Processed {} of {} events.\n".format(processed, count))
+        try:
+            eventstream.insert(
+                group=event.group,
+                event=event,
+                is_new=False,
+                is_regression=False,
+                is_new_group_environment=False,
+                primary_hash=primary_hash,
+                skip_consume=True,
+            )
+            processed += 1
+        except Exception as error:
+            print(
+                "An error occured while trying to instert the following event: {}\n.----\n{}".format(
```

Comment:
You've now introduced the possibility of this causing a UnicodeDecodeError. Make sure to always  use `unicode.format` instead of `str.format`.

So doing like, `f"thing {}".format(...)` instead.

Reply:
Well, `black` doesn't allow me to do `u"something {}".format()`. Suggestions?

Reply:
Yes it does. We use it everywhere. I borderline have a lint rule to prevent `str.format` since it's a major source of bugs.

I can only speculate you're doing something wrong.

Reply:
The file has `from __future__ import unicode_literals` at the top so that's why this wasn't a problem in the first place and that's why `black` was stripping the prefixes. Resolving this convo.